### ISSUE TITLE
Using %2f in custom routes returns 404 when deployed to Azure

### ISSUE THREAD

**Issue by hannesne** [Azure/azure-functions-host#3998]
#3916 Is supposed to fix using a + character in custom routes. 

This problem still exists in version 2.0.12275.0 of the runtime host when using the "/" character, escaped to %2f. 

For instance, I have a function that returns a message specified in the route. 
Locally calling `http://localhost:7071/api/Echo/my%2fmessage` works fine. When deployed to Azure `http://hannesechocsv2.azurewebsites.net/api/Echo/my%2fmessage` returns a 404.

The code I've deployed is available in https://github.com/hannesne/EchoFunction/tree/RoutingSample/CSharp/V2  

This seems similar to Azure/Azure-Functions#1949

**Comment by brandonh-msft** [comment]
ping @jeffhollan @fabiocav

**Comment by jeffhollan** [comment]
@soninaren any thoughts on this?  The fact that it works locally but not when deployed is interesting as well. Not sure if that was the case with #3916

**Comment by bkalafat** [comment]
I have same problem too. Can't access service with string which contains %2F.

**Comment by JanDonnermayer** [comment]
I encounter the exact same issue. Works when running locally, but not when deployed

**Comment by JanDonnermayer** [comment]
Interestingly, the problem can be resolved by double-escaping the string, that is, applying the escaping recursively two times.
This works both locally and deployed,

**Comment by SriGandla** [comment]
Yes this worked for me if i added %252f in the Function App url as opposed to %2f when i invoked the function locally.

**Comment by apatriz** [comment]
Even double-escaping does not work when using an Azure Function Proxy (v2). Any insight on this ?

**Comment by mindraptor** [comment]
I'm surprised this is still happening. I can confirm the double escaping %252f does work. Shouldn't this be fixed?

**Comment by naijith** [comment]
Facing the same issue with Azure web app. %252f seems to be working but can't change the encoding as it's in a library.

**Comment by alateefah** [comment]
Same issue. Double encoding worked for me, like this, `encodeURIComponent(encodeURIComponent("https://some-uri.com"))`

**Comment by MitchBodmer** [comment]
#https://github.com/Azure/azure-functions-host/issues/9290

**Comment by sgerace** [comment]
This is affecting us as well. Really annoying as clearly there is some additional layer of processing that is occurring on the Azure side of things that is hidden from developers.

**Comment by netscout** [comment]
Still no answers? 😰

**Comment by MitchBodmer** [comment]
FYI there has been acknowledgement on the other bug Azure/azure-functions-host#9290.

@mattchenderson assuming this is the host repo you were talking about in your comment, this might be the right bug to use.
